Synview-frontend/src/services/apiSlice.ts

```diff
@@ -56,6 +60,29 @@ export const apiSlice = createApi({
     }),
     getMyUpdates: builder.query<Updates, string>({
       query: (id) => `getMyUpdates/${id}`,
+      async onCacheEntryAdded(
+        arg,
+        { updateCachedData, cacheDataLoaded, cacheEntryRemoved }
+      ) {
+        const ws = new WebSocket(wsurl);
+        try {
+          await cacheDataLoaded;
+          const listener = (event: MessageEvent) => {
+            const data = JSON.parse(event.data);
+            if (!PostUpdateSchema.safeParse(data).success) {
+              throw new Error("Schema didnt validate correctly");
+            }
+            updateCachedData((draft) => {
+              draft.push(data);
+            });
+          };
+          ws.addEventListener("message", listener);
+        } catch {
+          throw new Error("Couldnt update updates trough websocket");
+        }
+        await cacheEntryRemoved;
+        ws.close();
```

Comment:
WebSockets are meant to be persistent ie. when app launches, it connects, and that connection stays open for the whole duration of the app lifetime. I am not sure this is the proper way to do this, let's chat tomorrow